segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -614,6 +624,37 @@ public boolean isOffline() {
     }
 
     //endregion
+    public CompletableFuture<SnapshotInfo> readStorageSnapshot(Duration timeout) {
+        val attributeIds = new ArrayList<UUID>();
+        attributeIds.add(ATTRIBUTE_SLTS_LATEST_SNAPSHOT_ID);
+        attributeIds.add(ATTRIBUTE_SLTS_LATEST_SNAPSHOT_EPOCH);
+        val segmentId =   this.metadata.getStreamSegmentId(NameUtils.getMetadataSegmentName(this.metadata.getContainerId()), false);
+        if (segmentId != ContainerMetadata.NO_STREAM_SEGMENT_ID) {
+            val map =  this.metadata.getStreamSegmentMetadata(segmentId).getAttributes();
+            val epoch = map.getOrDefault(ATTRIBUTE_SLTS_LATEST_SNAPSHOT_EPOCH, 0L);
+            val snapshotId = map.getOrDefault(ATTRIBUTE_SLTS_LATEST_SNAPSHOT_ID, 0L);
+            if (epoch > 0) {
+                val retValue = SnapshotInfo.builder()
+                        .snapshotId(snapshotId)
+                        .epoch(epoch)
+                        .build();
+                log.info("{}: Read SLTS snapshot. {}", this.traceObjectId, retValue);
+                return CompletableFuture.completedFuture(retValue);
+            }
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @SneakyThrows
+    public CompletableFuture<Void> saveStorageSnapshot(SnapshotInfo checkpoint, Duration timeout) {
+        TimeoutTimer timer = new TimeoutTimer(timeout);
+        val attributeUpdates = new ArrayList<AttributeUpdate>();
+        attributeUpdates.add(new AttributeUpdate(ATTRIBUTE_SLTS_LATEST_SNAPSHOT_ID, Replace, checkpoint.getSnapshotId()));
```

Comment:
same here. `Arrays.asList` is slightly more efficient as it doesn't create a real list; it just wraps an array.

Reply:
done.